app/models/planning_application.rb

```diff
@@ -560,4 +569,16 @@ def determination_date_is_not_in_the_future
       errors.add(:determination_date, "Determination date must be today or in the past")
     end
   end
+
+  def set_application_number
+    self.application_number = local_authority.planning_applications.count + 100
+  end
+
+  def created_at_year
+    created_at.year % 100
```

Comment:
How about `created_at.strftime("%y")` which is a [ruby method](https://apidock.com/ruby/DateTime/strftime) and does that